quicksilver-utils-async/src/desktop/websocket.rs

```diff
@@ -0,0 +1,100 @@
+
+use url::Url;
+use futures_io::{AsyncRead, AsyncWrite};
+
+use async_std::net::TcpStream;
+use async_tls::TlsConnector;
+use soketto::{
+    handshake::{Client, ServerResponse, Error as HandshakeError},
+    connection::{Sender, Receiver, Error as ConnectionError}
+};
+use std::io::{Error as IoError};
+use std::sync::Arc;
+use std::cell::RefCell;
+
+use log::{debug};
+
+use crate::{websocket::{WebSocketError, WebSocketMessage}};
+
+#[derive(Clone)]
+pub struct AsyncWebSocket {
+    sender: Arc<RefCell<Sender<Box<dyn AsyncStream>>>>,
+    receiver: Arc<RefCell<Receiver<Box<dyn AsyncStream>>>>,
+}
+
+impl From<HandshakeError> for WebSocketError {
+    fn from(err: HandshakeError) -> Self {
+        WebSocketError::NativeError(format!("Handshake error: {}", err))
+    }
+}
+
+impl From<ConnectionError> for WebSocketError {
+    fn from(err: ConnectionError) -> Self {
+        WebSocketError::NativeError(format!("Connection error: {}", err))
+    }
+}
+
+impl From<IoError> for WebSocketError {
+    fn from(err: IoError) -> Self {
+        WebSocketError::NativeError(format!("IO Error: {}", err))
+    }
+}
+
+trait AsyncStream: AsyncRead + AsyncWrite + Unpin {}
```

Comment:
this feels uneccessary... should be able to inline this combination of traits, right?